salt/states/git.py

```diff
@@ -1308,6 +1308,24 @@ def latest(name,
                         'if it does not already exist).',
                         comments
                     )
+                remote_tags = set([
+                    x.split('/')[-1] for x in __salt__['git.ls_remote'](
+                        cwd=target,
+                        remote=remote,
+                        opts="--tags",
+                        user=user,
+                        password=password,
+                        identity=identity,
+                        saltenv=__env__,
+                        ignore_retcode=True,
+                    ).keys() if '^{}' not in x
+                ])
+                if set(all_local_tags) != remote_tags:
+                    has_remote_rev = False
+                    ret['changes']['tags'] = {
+                        'old': all_local_tags,
+                        'new': list(remote_tags)
```

Comment:
The only thing I'm uneasy about here is putting all the tags in the repo in the changes dict. For repos with a lot of tags, this would produce a _lot_ of output. What about just including the new tags in a key under `ret['changes']`